### PR TITLE
armagetronad: 0.2.9.1.0 -> 0.2.9.1.1 + refactor

### PR DESCRIPTION
**Pull request by numinit**
## Description of changes

Update to latest and fix potential crash due to --enable-memmanager (#263244). Add an installCheckPhase and the following passthrus for obscure types of game servers:

- `pkgs.armagetronad`
- `pkgs.armagetronad.dedicated`
- `pkgs.armagetronad."0.2.9-sty+ct+ap"`
- `pkgs.armagetronad."0.2.9-sty+ct+ap".dedicated`
- `pkgs.armagetronad."0.4"`
- `pkgs.armagetronad."0.4".dedicated`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/armagetronad/default.nix [hunk 1]
```diff
@@ -1,42 +1,77 @@
...
+      nativeBuildInputs = [ pkg-config ];
+
+      buildInputs = [ libxml2 zlib ]
+        ++ lib.optionals (!dedicatedServer) [ SDL SDL_image libxml2 libjpeg libpng libGLU libGL ];
+
+      # Just use make all.
+      buildPhase = "make -j";
+
+      # Don't parallelize this to avoid a race.
+      installPhase = "make -j1 install";
```
Comment (numinit) [inline pkgs/games/armagetronad/default.nix:50]: Is there a nicer way to ask stdenv to enable parallel building but not for make install?

Comment (wegank) [reply pkgs/games/armagetronad/default.nix:120]:
```nix
  enableParallelBuilding = true;
  enableParallelInstalling = false;
```

Comment (numinit) [reply pkgs/games/armagetronad/default.nix:120]: Too easy. Thanks!

**Comment by numinit** [comment]
Marking as a draft until I hear back from the Armagetron team about which repos we should be using.

**Comment by numinit** [comment]
Switched from bzr to git per Armagetron team recommendation.

**Review by agbrooks — APPROVED** [review]

**Comment by numinit** [comment]
This is ready for merge.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 263675` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><code>armagetronad</code></li>
    <li><code>armagetronad-dedicated</code></li>
    <li><code>armagetronad."0.2.9-sty+ct+ap"</code> <i>(run manually, not recursed into)</i></li>
    <li><code>armagetronad."0.2.9-sty+ct+ap".dedicated</code> <i>(run manually, not recursed into)</i></li>
    <li><code>armagetronad."0.4"</code> <i>(run manually, not recursed into)</i></li>
    <li><code>armagetronad."0.4".dedicated</code> <i>(run manually, not recursed into)</i></li>
  </ul>
</details>

Seems to run fine, LGTM. Thanks

**Pull request merged**